src/Microsoft.AspNet.Identity/UserManager.cs

```diff
@@ -48,27 +48,23 @@ public class UserManager<TUser> : IDisposable where TUser : class
         /// <param name="contextAccessor">The accessor used to access the <see cref="HttpContext"/>.</param>
         public UserManager(IUserStore<TUser> store,
             IOptions<IdentityOptions> optionsAccessor,
-            IPasswordHasher<TUser> passwordHasher,
-            IEnumerable<IUserValidator<TUser>> userValidators,
-            IEnumerable<IPasswordValidator<TUser>> passwordValidators,
-            ILookupNormalizer keyNormalizer,
-            IdentityErrorDescriber errors,
             IServiceProvider services,
-            ILogger<UserManager<TUser>> logger,
-            IHttpContextAccessor contextAccessor)
+            ILogger<UserManager<TUser>> logger)
         {
             if (store == null)
             {
                 throw new ArgumentNullException(nameof(store));
             }
             Store = store;
-            Options = optionsAccessor?.Value ?? new IdentityOptions();
-            _context = contextAccessor?.HttpContext;
-            PasswordHasher = passwordHasher;
-            KeyNormalizer = keyNormalizer;
-            ErrorDescriber = errors;
             Logger = logger;
+            Options = optionsAccessor?.Value ?? new IdentityOptions();
+
+            _context = services?.GetService<IHttpContextAccessor>()?.HttpContext;
+            PasswordHasher = services?.GetService<IPasswordHasher<TUser>>() ?? new PasswordHasher<TUser>();
+            KeyNormalizer = services?.GetService<ILookupNormalizer>() ?? new UpperInvariantLookupNormalizer(); ;
+            ErrorDescriber = services?.GetService<IdentityErrorDescriber>() ?? new IdentityErrorDescriber(); ;
 
+            var userValidators = services?.GetService<IEnumerable<IUserValidator<TUser>>>();
             if (userValidators != null)
```

Comment:
In the old version of the code I don't think this null check was necessary. Our IoC containers are required to pass an empty IEnumerable to the constructor when no matching services are registered.